authentication/models.py

```diff
@@ -1,3 +1,50 @@
+from django.contrib.auth.models import AbstractUser
+from django.contrib.auth.base_user import BaseUserManager
 from django.db import models
+from django.utils.translation import gettext_lazy as _
 
-# Create your models here.
+
+class UserManager(BaseUserManager):
+    """Custom use class"""
+    use_in_migrations = True
+
+    def _create_user(self, email, password, **extra_fields):
+        """inner method to create user"""
+        if not email:
+            raise ValueError("Email field is required")
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_user(self, email, password=None, **extra_fields):
+        """outer method to create user. this method calls the _create_user
+            to create the user after setting the permission to none ."""
+        extra_fields.setdefault("is_staff", False)
+        extra_fields.setdefault("is_superuser", False)
+        return self._create_user(email, password, **extra_fields)
+
+    def create_superuser(self, email, password, **extra_fields):
+        """method create superuser"""
+        extra_fields.setdefault("is_staff", True)
+        extra_fields.setdefault("is_superuser", True)
+
+        if extra_fields.get("is_staff") is not True:
+            raise ValueError("Superuser must have is_staff=True.")
+        if extra_fields.get("is_superuser") is not True:
+            raise ValueError("Superuser must have is_superuser=True")
+        return self._create_user(email, password, **extra_fields)
+
+
+class User(AbstractUser):
+    username = None # not going to use username field
```

Comment:
@DamaMichaelYohanna why did you remove the username field?

Reply:
@Teams-Motion Because we are going to be handling authentication via email and not user name

Reply:
yes I know but will authenticate users via authenticate(request, email=email, password=password) and not authenticate(request, username=username, password=password)

Reply:
We're just changing to username to point to email instead

Reply:
@DamaMichaelYohanna  kindly  add the username field back. Delete the branch and make and another pull request. so i can merge it now. Thank you

Reply:
Alright

Reply:
Thank you @DamaMichaelYohanna

Reply:
@DamaMichaelYohanna thank you

Reply:
@Teams-Motion Done

Reply:
@DamaMichaelYohanna. Good I'm going to test it before merging it to the main branch. You made the username field unique? it prefer having the email as unique.

Reply:
@Teams-Motion the point is do we really need the username field? Remember the class we inherited from will provide us with first name and last name. What's the point having username since we are not authenticating with it?

Reply:
@DamaMichaelYohanna I would have liked us to over-ride that and use only the user. Because during sing up we will require only Email, username and password. Then we can easily reach out to our users via request.user.username. Do you understand?

Reply:
@Teams-Motion yes I do. I will take care of it when am back on sit.

Reply:
@DamaMichaelYohanna  okay. Thanks

Reply:
@Teams-Motion I have removed the first and last name fields.

Reply:
okay let me test it rigth away @DamaMichaelYohanna